src/service/git/base-git-api-client.ts

```diff
@@ -0,0 +1,60 @@
+import { EventData } from "@bc/domain/configuration";
+import { GitTokenService } from "@bc/service/git/git-token-service";
+import { BaseLoggerService } from "@bc/service/logger/base-logger-service";
+import { LoggerService } from "@bc/service/logger/logger-service";
+import Container from "typedi";
+
+type Response<T> = {
+  status: number;
+  data: T;
+};
+
+export abstract class BaseGitAPIClient {
+  protected tokenService: GitTokenService;
+  protected tokenServiceId: string;
+  protected baseUrl: string;
+  protected logger: BaseLoggerService;
+
+  constructor(baseUrl: string, tokenServiceId: string) {
+    this.tokenService = Container.get(GitTokenService);
+    this.tokenServiceId = tokenServiceId;
+    this.baseUrl = baseUrl;
+    this.logger = Container.get(LoggerService).logger;
+  }
+
+  abstract get repos(): {
+    getBranch: (args: {
```

Comment:
does it worth to extract the returned object structures in a dedicated types file? I think it could be easier to read the BaseGitAPIClient API if we have defined return interfaces/types, wdyt?

Reply:
Yes I was debating about this too but I wasn't sure if it was worth adding a separate file for the types cause I am not using them anywhere else and the abstract class is fairly small. But in retrospect, I think it will be nice to break these down and have access to types of argument and returned object and reuse that in the concrete classes. It will make creating new clients easier

Reply:
+1 I agree, more for this `It will make creating new clients easier`